### PR TITLE
refactor(ci): Completely remove stage that uploads telemetry when it's not needed, instead of skipping it

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

I noticed that this stage still "runs" for PR builds in the public ADO project and is always skipped because part of the condition (`BuildReason == IndividualCI`) is never true there. Feels cleaner to just not create that stage at all in that case. This makes it easier to tell at a glance if the `Build - client packages` runs in the public project ran successfully to completion (no need to check what the skipped part is).

The [pipeline run](https://dev.azure.com/fluidframework/public/_build/results?buildId=212504&view=results) for this PR already doesn't have the conditional stage "Upload pipeline run telemetry to Kusto":

image

Compared to current runs:

image

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

I suggest using the "Hide whitespace" option in the Github diff view; the changes are fairly minimal and easier to read without all the extra indentation that came from the additional `${{ if ... }}` block.

### CODE REVIEW

**Comment by tylerbutler** [comment]
@alexvy86 unrelated to this PR... these bundle size differences of just a few bytes is worrisome. I've seen it on a few different PRs now. I wonder if it's related to the build changes I've been making. I'm out for a couple of weeks so won't be able to investigate myself until December. It can probably wait until then.

**Comment by tylerbutler** [comment]
> @alexvy86 unrelated to this PR... these bundle size differences of just a few bytes is worrisome. I've seen it on a few different PRs now. I wonder if it's related to the build changes I've been making. I'm out for a couple of weeks so won't be able to investigate myself until December. It can probably wait until then.

@RishhiB FYI the bundle size issues might be related to the build changes I made/you're making.

File: tools/pipelines/build-client.yml [hunk 1]
```diff
@@ -138,7 +138,7 @@ extends:
     taskLint: false
     taskBuildDocs: true
     publishDocs: true
-    telemetry: true
+    telemetry: ${{ eq(variables['Build.Reason'], 'IndividualCI') }}
```
Comment (Josmithr) [inline tools/pipelines/build-client.yml:141]: Can we add a brief comment explaining this choice?

File: tools/pipelines/templates/build-npm-package.yml [hunk 2]
```diff
@@ -560,44 +560,45 @@ stages:
...
-  - stage: upload_run_telemetry
-    displayName: Upload pipeline run telemetry to Kusto
-    condition: and(succeededOrFailed(), eq(variables['Build.Reason'], 'IndividualCI'), eq(${{parameters.telemetry}}, true))
-    dependsOn:
-      - build
+  # Capture pipeline stage results
+  - ${{ if eq(parameters.telemetry, true) }}:
+    - stage: upload_run_telemetry
+      displayName: Upload pipeline run telemetry to Kusto
+      condition: succeededOrFailed()
```
Comment (Josmithr) [inline tools/pipelines/templates/build-npm-package.yml:567]: This indentation change is interesting. I'm surprised the previous indentation didn't cause issues.

**Review by Josmithr — APPROVED** [review]

File: tools/pipelines/templates/build-npm-package.yml (hunk 2 above)
Comment (alexvy86) [reply tools/pipelines/templates/build-npm-package.yml:567]: I was also surprised. YAML parsing and validation feels like black magic every time we see one of these.

**Pull request merged**